sql/core/src/test/scala/org/apache/spark/sql/internal/CatalogSuite.scala

```diff
@@ -743,4 +743,25 @@ class CatalogSuite extends SharedSparkSession with AnalysisTest with BeforeAndAf
     val catalogName2 = "catalog_not_exists"
     assert(!spark.catalog.databaseExists(Array(catalogName2, dbName).mkString(".")))
   }
+
+  test("three layer namespace compatibility - get database") {
+    Seq(("testcat", "somedb"), ("testcat", "ns.somedb")).foreach { case (catalog, dbName) =>
+      val qualifiedDb = s"$catalog.$dbName"
+      // TODO test properties? WITH DBPROPERTIES (prop='val')
+      sql(s"CREATE NAMESPACE $qualifiedDb COMMENT 'test comment' LOCATION '/test/location'")
+      val db = spark.catalog.getDatabase(qualifiedDb)
+      assert(db.name === dbName)
+      assert(db.description === "test comment")
+      assert(db.locationUri === "file:/test/location")
+    }
+    intercept[AnalysisException](spark.catalog.getDatabase("randomcat.db10"))
+  }
+
+  test("get database when there is `default` catalog") {
+    spark.conf.set("spark.sql.catalog.default", classOf[InMemoryCatalog].getName)
+    val db = "testdb"
+    val qualified = s"default.$db"
+    sql(s"CREATE NAMESPACE $qualified")
```

Comment:
I think what we should test is: there is a database called `testdb` in Hive Metastore, and also a namespace called `testdb` in `testcat`. The current catalog is `testcat` (we can change it via `sql("SET CATALOG testcat")`), and `getDatabase` should return the database in Hive Metastore. We can check the catalog by `Database.catalog`.

Reply:
I've implemented this test but currently just verifying based on setting the 'comment' of the Database. Let me know if there is some `Database.catalog` API i am not aware of